src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+                - regularizer_strength (`float`, *optional*, defaults to `0.02`): Strength of the regularizer term that
+                  pulls the solution to the unit range.
+                - max_iter (`int`, *optional*, defaults to `2`): Number of numerical optimizer function evaluations.
+                - tol (`float`, *optional*, defaults to `1e-3`): Numerical tolerance of the optimizer.
+                - max_res (`int`, *optional*, defaults to `None`): Resolution at which the search of scale and shift
+                  parameters is performed; `None` matches the `processing_resolution`.
+            input_latent (`torch.Tensor`, or `List[torch.Tensor]`, *optional*, defaults to `None`):
+                Latent noise tensors to replace the random initialization. These can be taken from the previous
+                function call's output.
+            generator (`torch.Generator`, or `List[torch.Generator]`, *optional*, defaults to `None`):
+                Random number generator object to ensure reproducibility.
+            output_prediction_format (`str`, *optional*, defaults to `"np"`):
+                Preferred format of the output's `prediction` and the optional `uncertainty` fields. The accepted
+                values are: `"np"` (numpy array) or `"pt"` (torch tensor).
+            output_visualization (`bool`, *optional*, defaults to `True`):
+                When enabled, the output's `visualization` field contains a PIL.Image that can be used for visual
+                quality inspection.
+            output_visualization_kwargs (`dict`, *optional*, defaults to `None`):
+                Extra dictionary with arguments for precise visualization control. The following options are available:
+                - color_map (`str`, *optional*, defaults to `"Spectral"`): Color map used to convert a single-channel
+                  depth prediction into colored representation.
+                - vis_min (`float`, *optional*, defaults to `0.0`): Minimum value of the visualized depth range.
+                - vis_max (`float`, *optional*, defaults to `1.0`): Maximum value of the visualized depth range.
+            output_uncertainty (`bool`, *optional*, defaults to `True`):
+                When enabled, the output's `uncertainty` field contains the predictive uncertainty map, provided that
+                the `ensemble_size` argument is set to a value above 2.
+            output_latent (`bool`, *optional*, defaults to `False`):
+                When enabled, the output's `latent` field contains the latent codes corresponding to the predictions
+                within the ensemble. These codes can be saved, modified, and used for subsequent calls with the
+                `input_latent` argument.
+
+        Examples:
+
+        Returns:
+            `MarigoldDepthOutput`: Output class instance for Marigold monocular depth prediction pipeline.
+        """
+
+        device = self._execution_device
+
+        # deprecations
```

Comment:
Do we need to have deprecations in the first place for a pipeline that's being _officially_ introduced in this PR? I don't think so.

Reply:
That's one more thing to discuss: what do we intend to do with the [original community pipeline](https://github.com/huggingface/diffusers/blob/main/examples/community/marigold_depth_estimation.py)? I thought that this new code could be used with existing codebases that had already adopted that pipeline. Of course, maintaining backward compatibility is not necessary. My initial plan that I wanted to discuss with you was to forward calls to the old pipeline into the new one, in which case these deprecation warnings would be triggered, suggesting the user to update their code (and discover new features and other pipelines).

Reply:
(agreed to remove all deprecation warnings from the normals pipeline)

Reply:
Removed all deprecations since the old pipeline will stay where it is